### PR TITLE
Fetch directly from Microsoft rather than using installed version of MSVC

### LINKED ISSUE #20: First steps towards vs2019 optional feedstock
A large package of mine (_10k lines of code_) fails when compiling with vs2017 with internal compiler errors, however it passes with vs2019. Other platforms pass as well. Following a discussion with @beckermr, he chaired a talking point at the `conda-forge/core` team meeting concerning making vs2019 a supported feedstock. The minutes recorded are as follow. 

>  - What do we need to do to get this done?
>    - _(@isuruf) Is this going to be global or just for a few feedstocks?_ If it's global, that's going to be a bit of a problem. you can link libs with 2017 and 2019 together, but you need 2019 to do the linking. This will require people building conda packages locally to update to vs2019. Uwe was cross-compiling but we only have vs2017.
>    - Which feedstocks need to be updated? Just the vc one
>    - new universal runtime which adds new DLLs
>      * not on windows 10
>      * can download it from windows update
>      * make a new runtime package
>      * @jjhelmus posted a note on the filename in gitter
>        * try and use paul's PR for vc for 2019
>        * do that PR for 2017 and try it on vc_dev channel
>      * then do for 2019
>      * keep track of where version numbers come from

### PR DESCRIPTION
**Pull request by nowster**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #20
Closes #18

### CODE REVIEW

**Comment by nowster** [comment]
Note that existing recipe does not package the license file, which is proprietary and most definitely not BSD as claimed.

**Comment by beckermr** [comment]
bump here @conda-forge/core

**Comment by beckermr** [comment]
bump @conda-forge/vc

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,64 +1,106 @@
...
+#  This is the version number reported by cl.exe
+{% set cl_version = "19.26.28806" %}
+
 package:
   name: vs{{ vsyear }}
-  version: {{ fullver }}
+  version: {{ runtime_version }}
+
+source:
+  - url: https://aka.ms/vs/16/release/vc_redist.x64.exe  # [win64]
```
Comment (isuruf) [inline recipe/meta.yaml:29]: Is this a stable link?

Comment (nowster) [reply recipe/meta.yaml:29]: Yes, that is the stable link which redirects to their CDN.

Comment (isuruf) [reply recipe/meta.yaml:29]: I mean, they update the actual file from time to time right? The file I get today might not be the same one I'll get in a few weeks right?

**Comment by nowster** [comment]
There's currently a problem building this due to `run_exports` pinning conflicts with previous installed versions of this package. I haven't been able to resolve them using the `ignore_run_exports` mechanisms.

File: recipe/meta.yaml [hunk 2]
```diff
@@ -21,15 +21,25 @@
...
   name: vs{{ vsyear }}
   version: {{ runtime_version }}
 
 source:
-  - url: https://aka.ms/vs/16/release/vc_redist.x64.exe  # [win64]
-    sha256: 7d7105c52fcd6766beee1ae162aa81e278686122c1e44890712326634d0b055e  # [win64]
-  - url: https://aka.ms/vs/16/release/vc_redist.x86.exe  # [win32]
-    sha256: a06aac66734a618ab33c1522920654ddfc44fc13cafaa0f0ab85b199c3d51dc0  # [win32]
+  - url: https://download.visualstudio.microsoft.com/download/pr/{w64_uuid}/{w64_sha}/vc_redist.x64.exe  # [win64]
+    sha256: { w64sha | lower }  # [win64]
```
Comment (dhirschfeld) [inline recipe/meta.yaml:38]:
```suggestion
    sha256: { w64_sha | lower }  # [win64]
```

**Comment by isuruf** [comment]
This looks good to me. Can we do VS2017 first though (or both VS2017 and VS2019)?

**Comment by traversaro** [comment]
Hi everyone, I am not sure if anyone is working on this, but I would be happy to give an hand to get this merged.

**Comment by isuruf** [comment]
Yes, it's the only issue right now. Thanks for taking a look

**Comment by traversaro** [comment]
I started looking into this, I have a few question before actually opening a PR: 
* For supporting both VS2017 and VS2019, does it make sense to use two separate branches as done in normal "feedstock" to generate different packages, or you had in mind another strategy? 
* There is any reason why 32-bit is still supported in this PR, or we can just drop support for it?

**Comment by isuruf** [comment]
Thanks @nowster

**Pull request merged**